magpie/alembic/versions/2020-09-22_a2a039e2cff5_explicit_permission_names.py

```diff
@@ -43,7 +43,7 @@ def upgrade():
     grp_res_perms = GroupResourcePermissionService.base_query(db_session=session)
     usr_res_perms = UserResourcePermissionService.base_query(db_session=session)
 
-    for perm_list in [grp_res_perms, usr_res_perms]:
+    for item_id_key, perm_list in [("group_id", grp_res_perms), ("user_id", usr_res_perms)]:
```

Comment:
I have no idea about DB migration but shouldn't the fix be in a new migration step?  Fixing in an existing migration step means a server that already execute this step won't attempt it again and miss out on the fix?

Discard what I said if it makes no sense since I know nothing about DB migration.

Reply:
It has to be applied there so that a server that did not run the migration will do it properly after rollback.
If added as new step, permissions will still be dropped at that step and later 'new' step will have no effect as there is nothing left to work with.
Migration steps have their own version dissociated from magpie versions.

Reply:
> It has to be applied there so that a server that did not run the migration will do it properly after rollback.

Oh wait so there is a way to rollback a DB migration?  I didn't have to `rm -rf /data/magpie_persist` when reverting from 3.5.1 to 1.7.3?